Description: Avoid test crash

First seen on numpy2 transition, unknown if actually related to that

Author: Matthew Roeschke
Origin: upstream pull 60416
Forwarded: not-needed

--- a/pandas/tests/test_algos.py
+++ b/pandas/tests/test_algos.py
@@ -1281,7 +1281,7 @@ class TestValueCounts:
             result_dt = algos.value_counts(dt)
         tm.assert_series_equal(result_dt, exp_dt)
 
-        exp_td = Series({np.timedelta64(10000): 1}, name="count")
+        exp_td = Series([1], index=[np.timedelta64(10000)], name="count")
         with tm.assert_produces_warning(FutureWarning, match=msg):
             result_td = algos.value_counts(td)
         tm.assert_series_equal(result_td, exp_td)
